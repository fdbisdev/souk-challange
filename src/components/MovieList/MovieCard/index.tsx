import React, { useEffect, useState } from 'react';

import { useNavigation } from '@react-navigation/native';

import {
  Container,
  MovieBanner,
  MovieDate,
  MovieName,
  MovieDateWrapper,
} from './styles';
import { Movie, MovieDetailsProps } from '../../../utils/interfaces';

function MovieCard({ movieName, movieDate, movieBanner, movieId }: Movie) {
  const [dateFormatted, setDateFormatted] = useState(movieDate);

  const navigation = useNavigation<MovieDetailsProps['navigation']>();

  function formatDate() {
    const dateSplit = movieDate.split('-');
    setDateFormatted(`${dateSplit[2]}/${dateSplit[1]}/${dateSplit[0]}`);
  }

  function handlePressMovieCard() {
    navigation.navigate('MovieDetails', {
      movieName,
      movieDate,
      movieId,
      movieBanner,
    });
  }

  useEffect(() => {
    formatDate();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Container onPress={() => handlePressMovieCard()}>
      <MovieBanner source={{ uri: movieBanner }}>
        <MovieDateWrapper>
          <MovieDate>{dateFormatted}</MovieDate>
        </MovieDateWrapper>
      </MovieBanner>
      <MovieName>{movieName}</MovieName>
    </Container>
  );
}

export default MovieCard;
