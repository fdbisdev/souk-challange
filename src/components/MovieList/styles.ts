import styled from 'styled-components/native';
import colors from '../../utils/colors';

export const Container = styled.View`
  flex: 1;
  margin-top: 5%;
  width: 100%;
  align-items: center;
  justify-content: center;
`;

export const Title = styled.Text`
  color: ${colors.white};
`;
