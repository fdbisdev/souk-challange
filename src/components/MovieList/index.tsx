/* eslint-disable react/jsx-no-bind */
import React, { useEffect, useState } from 'react';
import { ActivityIndicator, FlatList } from 'react-native';
import { getPopularMovies, getUpcomingMovies } from '../../services/api';
import colors from '../../utils/colors';
import { IMovie, MovieArrayListProps } from '../../utils/interfaces';
import MovieCard from './MovieCard';

import { Container } from './styles';

function MovieList({ popular, upcoming }: IMovie) {
  const [upcomingMoviesList, setUpcomingMoviesList] = useState();
  const [popularMoviesList, setPopularMoviesList] = useState();
  const [loading, setLoading] = useState(true);

  const getMovieBannerBaseURL = 'https://image.tmdb.org/t/p/w500/';

  async function getMovieList() {
    const popularMoviesResponse: any = await getPopularMovies();
    setPopularMoviesList(popularMoviesResponse.data.results);

    const upcomingMoviesResponse: any = await getUpcomingMovies();
    setUpcomingMoviesList(upcomingMoviesResponse.data.results);

    setLoading(false);
  }

  useEffect(() => {
    getMovieList();
  }, []);

  if (loading) {
    return (
      <Container>
        <ActivityIndicator size="large" color={colors.white} />
      </Container>
    );
  }

  // eslint-disable-next-line react/no-unstable-nested-components
  function CardList({ item }: { item: MovieArrayListProps }) {
    return (
      <MovieCard
        movieBanner={getMovieBannerBaseURL + item.poster_path}
        movieDate={item.release_date}
        movieName={item.title}
        movieId={item.id}
      />
    );
  }

  return (
    <Container testID="movieList">
      {popular && !upcoming ? (
        <FlatList
          testID="popularList"
          data={popularMoviesList}
          keyExtractor={item => String(item.id)}
          numColumns={3}
          renderItem={CardList}
        />
      ) : (
        <FlatList
          testID="upcomingList"
          data={upcomingMoviesList}
          numColumns={3}
          keyExtractor={item => String(item.id)}
          renderItem={CardList}
        />
      )}
    </Container>
  );
}

export default MovieList;
