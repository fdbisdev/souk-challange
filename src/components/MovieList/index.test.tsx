import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import MovieList from '.';

describe('Movie List Component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<MovieList popular={false} upcoming />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
