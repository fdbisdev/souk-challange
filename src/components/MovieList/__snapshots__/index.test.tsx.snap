// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List Component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginTop": "5%",
        "width": "100%",
      },
    ]
  }
>
  <ActivityIndicator
    color="#ffffff"
    size="large"
  />
</View>
`;
